### PR TITLE
Repositories and Create Movie Cards

### PR DESCRIPTION
**Pull request by mikolaslorant**
## Summary
-Added  and implemented UnitOfWork and Repository patterns.
-Added movie list
-Added movie creation
-new clean migration with Id column instead of ID
## Trello card
-Creado
-Repositories
## Screenshot
screenshot from 2018-08-24 18-12-51
screenshot from 2018-08-24 18-13-09

### CODE REVIEW

File: src/Controllers/HelloWorldController.cs [hunk 1]
```diff
@@ -7,27 +7,44 @@
...
+using MlNetCore.Models.Views;
 
 namespace MlNetCore.Controllers
 {
     public class HelloWorldController : Controller
     {
         private readonly IHtmlLocalizer<HelloWorldController> _localizer;
-        
-        public HelloWorldController(IHtmlLocalizer<HelloWorldController> localizer)
+        private readonly IUnitOfWork _unitOfWork;
```
Comment (gussiciliano) [inline src/Controllers/HelloWorldController.cs:19]: add a get method for this property

File: src/Controllers/HelloWorldController.cs [hunk 2]
```diff
@@ -7,27 +7,44 @@
...
+        public HelloWorldController(IUnitOfWork unitOfWork, IHtmlLocalizer<HelloWorldController> localizer)
         {
+            this._unitOfWork = unitOfWork;
             this._localizer = localizer;
         }
 
         public IActionResult Index()
         {
-            return View();
+            return View(_unitOfWork.MovieRepository.GetAll().ToList());
```
Comment (gussiciliano) [inline src/Controllers/HelloWorldController.cs]: use always the get method on this cases

File: src/Controllers/HelloWorldController.cs [hunk 3]
```diff
@@ -7,27 +7,44 @@
...
+
+        public IActionResult NewMovie()
+        {
+            return View(new MovieViewModel());
+        }
+        public IActionResult CreateNewMovie(MovieViewModel viewMovie)
+        {
+            Movie movieModel = new Movie(viewMovie.Id, viewMovie.Title, viewMovie.ReleaseDate, 
+                                        viewMovie.Genre, viewMovie.Price);
+            _unitOfWork.MovieRepository.Add(movieModel);
```
Comment (gussiciliano) [inline src/Controllers/HelloWorldController.cs]: use the get method on this and the next line

File: src/Repositories/Interfaces/IMovieRepository.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+using MlNetCore.Models;
+
+namespace MlNetCore.Repositories.Interfaces
+{
+    public interface IMovieRepository : IRepository<Movie>
+    {
+        Movie GetMovie(int id);
```
Comment (gussiciliano) [inline src/Repositories/Interfaces/IMovieRepository.cs]: this is not necessary, you can use the Get method on Repository class

File: src/Repositories/MovieRepository.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        public MovieRepository(DbContext context) : base(context)
+        {
+        }
+
+        public DataBaseContext MovieContext
+        {
+            get { return Context as DataBaseContext; }
+        }
+
+        public Movie GetMovie(int id)
```
Comment (gussiciliano) [inline src/Repositories/MovieRepository.cs]: same as before

File: src/Views/HelloWorld/index.cshtml [hunk 6]
```diff
@@ -1,6 +1,46 @@
...
+            <td>
+                <a asp-action="Edit" asp-route-id="@item.Id">Edit</a> |
+                <a asp-action="Details" asp-route-id="@item.Id">Details</a> |
+                <a asp-action="Delete" asp-route-id="@item.Id">Delete</a>
+            </td>
+        </tr>
+}
+    </tbody>
+</table>
+
```
Comment (gussiciliano) [inline src/Views/HelloWorld/index.cshtml]: add eof

**Review by gussiciliano — CHANGES_REQUESTED** [review]
just a few comments

File: src/Views/HelloWorld/index.cshtml (hunk 6 above)
Comment (mikolaslorant) [reply src/Views/HelloWorld/index.cshtml]: Already added, but does not show up in github.

**Comment by mikolaslorant** [comment]
Already corrected requested changes.

File: src/Controllers/HelloWorldController.cs [hunk 7]
```diff
@@ -16,7 +16,7 @@ namespace MlNetCore.Controllers
     public class HelloWorldController : Controller
     {
         private readonly IHtmlLocalizer<HelloWorldController> _localizer;
-        private readonly IUnitOfWork _unitOfWork;
+        private IUnitOfWork _unitOfWork { get; }
```
Comment (gussiciliano) [inline src/Controllers/HelloWorldController.cs]: change this for
`private readonly IUnitOfWork _unitOfWork;`

Comment (gussiciliano) [reply src/Controllers/HelloWorldController.cs]: and add this method after the Constructor method:
`private IUnitOfWork UnitOfWork
        {
            get { return this._unitOfWork; }
        }`

Comment (gussiciliano) [reply src/Controllers/HelloWorldController.cs]: this is the convention :P

File: src/Controllers/HelloWorldController.cs [hunk 8]
```diff
@@ -25,7 +25,7 @@ public HelloWorldController(IUnitOfWork unitOfWork, IHtmlLocalizer<HelloWorldCon
 
         public IActionResult Index()
         {
-            return View(_unitOfWork.MovieRepository.GetAll().ToList());
+            return View(this._unitOfWork.MovieRepository.GetAll().ToList());
```
Comment (gussiciliano) [inline src/Controllers/HelloWorldController.cs]: use `return View(UnitOfWork.MovieRepository.GetAll().ToList());`

File: src/Controllers/HelloWorldController.cs [hunk 9]
```diff
@@ -42,8 +42,8 @@ public IActionResult CreateNewMovie(MovieViewModel viewMovie)
         {
             Movie movieModel = new Movie(viewMovie.Id, viewMovie.Title, viewMovie.ReleaseDate, 
                                         viewMovie.Genre, viewMovie.Price);
-            _unitOfWork.MovieRepository.Add(movieModel);
-            _unitOfWork.Complete();
+            this._unitOfWork.MovieRepository.Add(movieModel);
```
Comment (gussiciliano) [inline src/Controllers/HelloWorldController.cs]: same as bofore en these two lines

File: src/Views/HelloWorld/index.cshtml (hunk 6 above)
Comment (gussiciliano) [reply src/Views/HelloWorld/index.cshtml]: dont worry

**Review by gussiciliano — CHANGES_REQUESTED** [review]
i added the convention :)

**Review by gussiciliano — APPROVED** [review]

**Pull request merged**